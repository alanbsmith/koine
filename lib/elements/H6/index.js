import styled from 'styled-components';
import { applyStyleModifiers } from 'styled-components-modifiers';

import fontWeights from '../../modifiers/fontWeights';

const H6 = styled.h6`
  font-size: 20px;
  margin: 0;
  ${applyStyleModifiers(fontWeights)}
`;

export default H6;
