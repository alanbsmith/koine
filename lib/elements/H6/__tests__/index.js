import React from 'react';
import H6 from '../index';
import renderer from 'react-test-renderer';

import { renderWithTheme } from '__tests__/helpers';


describe('H6 Element', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <H6>Header Element</H6>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies lightWeight modifier', () => {
    const tree = renderWithTheme(
      <H6 modifiers={['lightWeight']}>Header Element</H6>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
