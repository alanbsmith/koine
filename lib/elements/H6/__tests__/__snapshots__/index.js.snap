// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H6 Element applies lightWeight modifier 1`] = `
<h6
  className="sc-bdVaJa gNceKw"
>
  Header Element
</h6>
`;

exports[`H6 Element renders correctly 1`] = `
<h6
  className="sc-bdVaJa bmfcQW"
>
  Header Element
</h6>
`;
